lib/order.rb

```diff
@@ -1,18 +1,74 @@
+require "csv"
+
 module Grocery
   class Order
-    attr_reader :id, :products
+    attr_reader :id, :products, :orders
+    @@orders = []
 
     def initialize(id, products)
       @id = id
       @products = products
-    end
+      if @products == nil
+        @products == 0
+      end
+    end #--- end initialize method
 
     def total
-      # TODO: implement total
-    end
+      total = 0
+      @products.each do |product, price|
+        total = total + price
+      end
+      total = total + (total * 0.075).round(2)
+      return total
+    end #--- end total method
 
     def add_product(product_name, product_price)
-      # TODO: implement add_product
-    end
-  end
-end
+      if @products.keys.include?(product_name)
+        return false
+      else
+        @products[product_name] = product_price
+        return true
+      end
+    end #--- end add_product method
+
+    def remove_product(product_name)
+      if @products.keys.include?(product_name)
+        @products.delete(product_name)
+        return true
+      else
+        return false
+      end
+    end #--- end remove_product method
+
+    def self.all
+      if @@orders.length == 0
```

Comment:
I like this idea - if we've already loaded the orders once, why do it again?

This technique is called "memoization", and it's a form of caching. There's potentially a big performance benefit here, because reading from disk is extremely expensive, but the downside is we don't have any way to pick up changes to the file made by a user or another program.